### PR TITLE
librarified! certify() can be called on its own

### PR DESCRIPTION
**Pull request by seanmonstar**
usage:

``` js
var certify = require('browserid-certifier').certify;

certify(options, callback);
```

@lloyd ohai!

### CODE REVIEW

File: .jshintrc [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    "exports", 
+    "require",
+    "process"
+  ],
+  "es5": true,
+  "esnext": true,
+  "shadow": false,
+  "supernew": false,
+  "strict": false
+}
```
Comment (lloyd) [inline .jshintrc:28]: bonus points for jshint.

File: lib/certify.js [hunk 2]
```diff
@@ -1,30 +1,31 @@
...
+ * Options:
+ * - pubkey
+ * - privkey
+ * - duration
+ * - email
+ * - hostname
+ */
+module.exports = function certify(options, callback) {
+  var pubKey = jwcrypto.loadPublicKey(options.pubkey);
+  var privKey = jwcrypto.loadSecretKey(options.privkey);
```
Comment (lloyd) [inline lib/certify.js:16]: out of scope of this change, but would be interesting to see how this fails and tighten up error handling - given that now this is lib-rarified more folks of less familiarity with persona are going to use it!

File: lib/certify.js [hunk 3]
```diff
@@ -1,30 +1,31 @@
...
+
+  var expiration = new Date();
+  var iat = new Date();
+
+  expiration.setTime(new Date().valueOf() + (options.duration * 1000));
+  // Set issuedAt to 10 seconds ago. Pads for verifier clock skew
+  iat.setTime(iat.valueOf() - (10 * 1000));
```
Comment (lloyd) [inline lib/certify.js:23]: again out of scope of this change, but would be useful to make this behavior modifiable via options with a default of 10s.

**Comment by lloyd** [comment]
r+, great incremental improvement.  I'll use this in mozilla idp immediately!

Things I'd like to see in the future:
1. unit tests which excercise the heck out of the programatic API
2. README update to document the two modes of usage.

:heart:

will publish to npm shortly

File: lib/certify.js (hunk 3 above)
Comment (seanmonstar) [inline lib/certify.js:23]: ya, as i was laying in bed thinking about this (of course), I thought that it'd be more powerful to create an Certifier instance, that has some initial setup properties, like hostname, the private key, clock skew, etc.

``` js
var certifier = new Certifier({
  hostname: 'foobar.com',
  privkey: PRIV_KEY,
  skew: 10 * 1000
});

//later...
certifier.certify({ pubkey: pubkey, principal: principal }, cb);
```

**Comment by seanmonstar** [comment]
I'd love those things too :)

**Pull request merged**